core/include/detray/geometry/detail/volume_descriptor.hpp

```diff
@@ -0,0 +1,158 @@
+    ///               shape, which is defined by its portals and are chosen in
+    ///               the detector builder.
+    constexpr volume_descriptor(const volume_id id,
+                                const array_t<scalar_t, 6> &bounds)
+        : _id(id), _bounds(bounds) {}
+
+    /// @returns the volume shape id, e.g. 'cylinder'
+    DETRAY_HOST_DEVICE
+    constexpr auto id() const -> volume_id { return _id; }
+
+    /// Set the volume bounds to @param bounds
+    DETRAY_HOST
+    constexpr void set_bounds(const array_t<scalar_t, 6> &bounds) {
+        _bounds = bounds;
+    }
+
+    /// @returns the volume bounds - const access
+    DETRAY_HOST_DEVICE
+    constexpr auto bounds() const -> const array_t<scalar_t, 6> & {
+        return _bounds;
+    }
+
+    /// @return the index of the volume in the detector volume container.
+    DETRAY_HOST_DEVICE
+    constexpr auto index() const -> dindex { return _index; }
+
+    /// @param index the index of the volume in the detector volume container.
+    DETRAY_HOST
+    constexpr auto set_index(const dindex index) -> void { _index = index; }
+
+    /// @returns link to all acceleration data structures - const access
+    DETRAY_HOST_DEVICE constexpr auto full_link() const -> const link_type & {
+        return _sf_finder_links;
+    }
+
+    /// @returns link for a specific type of object - const access.
+    template <ID obj_id>
+    DETRAY_HOST_DEVICE constexpr auto link() const -> const link_t & {
+        return detail::get<obj_id>(_sf_finder_links);
+    }
+
+    /// @returns link for a specific type of object - non-const access.
+    template <ID obj_id>
+    DETRAY_HOST_DEVICE constexpr auto link() -> link_t & {
+        return detail::get<obj_id>(_sf_finder_links);
+    }
+
+    /// Set surface finder during detector building
+    template <ID obj_id>
+    DETRAY_HOST constexpr auto set_link(const link_t &link) -> void {
+        _sf_finder_links[obj_id] = link;
+    }
+
+    /// sSt surface finder during detector building
+    template <ID obj_id>
+    DETRAY_HOST constexpr auto set_link(const typename link_t::id_type id,
+                                        const typename link_t::index_type index)
+        -> void {
+        _sf_finder_links[obj_id] = link_t{id, index};
+    }
+
+    /// Equality operator
+    ///
+    /// @param rhs is the right-hand side to compare against.
+    DETRAY_HOST_DEVICE
+    constexpr auto operator==(const volume_descriptor &rhs) const -> bool {
+        return (_bounds == rhs._bounds && _index == rhs._index &&
+                _sf_finder_links == rhs._sf_finder_links &&
+                _sf_finder_links[ID::e_sensitive] ==
+                    rhs._sf_finder_links[ID::e_sensitive]);
+    }
+
+    private:
+    /// How to interpret the boundary values
+    volume_id _id = volume_id::e_cylinder;
+
+    /// Boundary values, default for cylinder volume
+    array_t<scalar_t, 6> _bounds = {0.f,
```

Comment:
Is the bound dimension always 6? And I think it would be nicer if we can have this information somewhere else

Reply:
Yes, it is. We are not using the bounds anywhere, as far as I can tell. Since I don't really know what to use them for, I am also not sure what to do with them, tbh.

Reply:
I just spoke to Andi and I think it can be removed now

Reply:
Removed. From the volume itself it will very difficult now to find out if a given position lies inside of it. But we discussed that in the future the volume grid should be used for this